app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt

```diff
@@ -20,435 +20,295 @@
-    membersToMention: List<Contact>,
-    onPingClicked: () -> Unit,
-    onShowSelfDeletionOption: () -> Unit,
-    showSelfDeletingOption: Boolean,
-    tempWritableImageUri: Uri?,
-    tempWritableVideoUri: Uri?
+    messageListContent: @Composable () -> Unit
 ) {
-    BoxWithConstraints {
-        val onSendButtonClicked = remember {
-            {
-                val expireAfter = (messageComposerState.messageComposeInputState as? MessageComposeInputState.Active)?.let {
-                    (it.type as? MessageComposeInputType.SelfDeletingMessage)
-                }?.selfDeletionDuration?.value
+    with(messageComposerState) {
+        when (messageComposerState.interactionAvailability) {
+            InteractionAvailability.BLOCKED_USER -> BlockedUserComposerInput(securityClassificationType)
+            InteractionAvailability.DELETED_USER -> DeletedUserComposerInput(securityClassificationType)
+            InteractionAvailability.NOT_MEMBER, InteractionAvailability.DISABLED ->
+                MessageComposerClassifiedBanner(
+                    securityClassificationType = securityClassificationType,
+                    paddingValues = PaddingValues(vertical = dimensions().spacing16x)
+                )
 
-                onSendTextMessage(
-                    SendMessageBundle(
-                        message = messageComposerState.messageComposeInputState.messageText.text,
-                        mentions = messageComposerState.mentions,
-                        quotedMessageId = messageComposerState.quotedMessageData?.messageId
-                    )
+            InteractionAvailability.ENABLED -> {
+                EnabledMessageComposer(
+                    messageComposerState = messageComposerState,
+                    messageListContent = messageListContent,
+                    onSendButtonClicked = ::sendMessage
                 )
-                messageComposerState.quotedMessageData = null
-                messageComposerState.setMessageTextValue(TextFieldValue(""))
             }
         }
+    }
+}
 
-        val onSendEditButtonClicked = remember {
-            {
-                (messageComposerState.messageComposeInputState as? MessageComposeInputState.Active)?.let {
-                    (it.type as? MessageComposeInputType.EditMessage)?.messageId
-                }?.let { originalMessageId ->
-                    onSendEditTextMessage(
-                        EditMessageBundle(
-                            originalMessageId = originalMessageId,
-                            newContent = messageComposerState.messageComposeInputState.messageText.text,
-                            messageComposerState.mentions,
-                        )
-                    )
+@Composable
+private fun EnabledMessageComposer(
+    messageComposerState: MessageComposerState,
+    messageListContent: @Composable () -> Unit,
+    onSendButtonClicked: () -> Unit
+) {
+    with(messageComposerState) {
+        Row {
```

Comment:
I think it should be a `Column` so that `ActiveMessageComposer` is shown under `SecurityClassificationBanner`, and now with `Row`, when we show `SecurityClassificationBanner` then `ActiveMessageComposer` is shown to the right and if this banner takes the whole width then nothing else will be visible 🤔